File: TaverneSpringBoot/src/main/java/soprajc/TaverneSpringBoot/model/fonctionnalitees/Intervention.java
package soprajc.TaverneSpringBoot.model.fonctionnalitees;

import java.time.LocalDate;
import java.time.LocalTime;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import soprajc.TaverneSpringBoot.model.comptes.Intervenant;
import soprajc.TaverneSpringBoot.model.inventaire.Bar;

@Entity
public class Intervention {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	private LocalTime hDebut;
	
	private LocalTime hFin;
	
	private LocalDate date;

	private String typeIntervention;

	@Column(nullable = true)
	private int coutIntervenant;
	@Column(nullable = true)
	private int prixClient;
	
	@Enumerated(EnumType.STRING)
	private StatutIntervention statut;
	
	
	@ManyToOne
	@JoinColumn(name = "id_intervenant", nullable = false)
	private Intervenant intervenant;
	
	@ManyToOne
	@JoinColumn(name = "id_bar", nullable = false)
	private Bar bar;
	
	
	public Intervention() {}


	public Intervention(LocalTime hDebut, LocalTime hFin, LocalDate date, String typeIntervention, int coutIntervenant,
			int prixClient, Intervenant intervenant, Bar bar) {
		this.hDebut = hDebut;
		this.hFin = hFin;
		this.date = date;
		this.typeIntervention = typeIntervention;
		this.coutIntervenant = coutIntervenant;
		this.prixClient = prixClient;
		this.intervenant = intervenant;
		this.bar = bar;
		this.statut = StatutIntervention.EnAttente;
	}



	public StatutIntervention getStatut() {
		return statut;
	}


	public void setStatut(StatutIntervention statut) {
		this.statut = statut;
	}

	@Override
	public String toString() {
		return "Intervention [id=" + id + ", hDebut=" + hDebut + ", hFin=" + hFin + ", typeIntervention="
				+ typeIntervention + ", coutIntervenant=" + coutIntervenant + ", prixClient=" + prixClient
				+ ", intervenant=" + intervenant + ", bar=" + bar + "]";
	}


	public Bar getBar() {
		return bar;
	}


	public void setBar(Bar bar) {
		this.bar = bar;
	}


	public String getTypeIntervention() {
		return typeIntervention;
	}


	public void setTypeIntervention(String typeIntervention) {
		this.typeIntervention = typeIntervention;
	}


	public int getId() {
		return id;
	}


	public void setId(int id) {
		this.id = id;
	}


	public LocalTime gethDebut() {
		return hDebut;
	}


	public void sethDebut(LocalTime hDebut) {
		this.hDebut = hDebut;
	}


	public LocalTime gethFin() {
		return hFin;
	}


	public void sethFin(LocalTime hFin) {
		this.hFin = hFin;
	}


	public int getCoutIntervenant() {
		return coutIntervenant;
	}


	public void setCoutIntervenant(int coutArtiste) {
		this.coutIntervenant = coutArtiste;
	}


	public int getPrixClient() {
		return prixClient;
	}


	public void setPrixClient(int prixClient) {
		this.prixClient = prixClient;
	}


	public Intervenant getIntervenant() {
		return intervenant;
	}


	public void setIntervenant(Intervenant intervenant) {
		this.intervenant = intervenant;
	}

	public LocalDate getDate() {
		return date;
	}

	public void setDate(LocalDate date) {
		this.date = date;
	}
	
	

	
	
}

File: Taverne/src/app/model/fonctionnalitees/intervention.ts
import { Intervenant } from './../comptes/intervenant';
import { StatutIntervention } from './statutIntervention';

export class Intervention {
  constructor(
    private _debut?: Date,
    private _fin?: Date,
    private _type?: string,
    private _coutIntervenant?: number,
    private _prixClient?: number,
    private _statut?: StatutIntervention,
    private _intervenant?: Intervenant,
    private _bar?: Bar
  ) {}

  set debut(value: Date | undefined) {
    this._debut = value;
  }

  set fin(value: Date | undefined) {
    this._fin = value;
  }

  set type(value: string | undefined) {
    this._type = value;
  }

  set coutIntervenant(value: number | undefined) {
    this._coutIntervenant = value;
  }

  set prixClient(value: number | undefined) {
    this._prixClient = value;
  }

  set statut(value: StatutIntervention | undefined) {
    this._statut = value;
  }

  set intervenant(value: Intervenant | undefined) {
    this._intervenant = value;
  }

  set bar(value: Bar | undefined) {
    this._bar = value;
  }

  get debut(): Date | undefined {
    return this._debut;
  }

  get fin(): Date | undefined {
    return this._fin;
  }

  get type(): string | undefined {
    return this._type;
  }

  get coutIntervenant(): number | undefined {
    return this._coutIntervenant;
  }

  get prixClient(): number | undefined {
    return this.prixClient;
  }

  get statut(): StatutIntervention | undefined {
    return this._statut;
  }

  get intervenant(): Intervenant | undefined {
    return this._intervenant;
  }

  get bar(): Bar | undefined {
    return this._bar;
  }
}
